templates/logo-swap-close.html

```diff
@@ -0,0 +1,163 @@
+  {% if blockable %}
+    <div class="block-snippet-button block-snippet-button-{{ snippet_id }}">
+    </div>
+  {% endif %}
+  {% if blockable or clickable %}
+    <div id="block-snippet-overlay-{{ snippet_id }}" class="block-snippet-overlay-{{ snippet_id }}">
+    </div>
+  {% endif %}
+  {% if icon_url %}<a href="{{ icon_url|safe }}">{% endif %}
+    <img class="icon" src="{{ icon }}" />
+  {% if icon_url %}</a>{% endif %}
+  <p>{{ text|safe }}</p>
+</div>
+<script>
+//<![CDATA[
+(function() {
+    var snippet = document.getElementById('snippet-{{ snippet_id }}');
+    var newLogo = '{{ replacement_logo }}';
+    snippet.addEventListener('show_snippet', function(e) {
+        var brandLogo = document.getElementById('brandLogo');
+        if (brandLogo.src !== undefined) {
+            brandLogo.src = newLogo;
+        } else {
+            brandLogo.style.backgroundImage = 'url(' + newLogo + ')';
+        }
+        brandLogo.classList.add('switch-{{ snippet_id }}');
+
+        {% if logo_url %}
+          var url = snippet.getAttribute('data-url');
+          var parent = brandLogo.parentElement;
+          var searchContainer = document.getElementById('searchContainer');
+          var link = document.createElement('a');
+          brandLogo = parent.replaceChild(link, brandLogo);
+          link.className = 'logo-link-{{ snippet_id }}';
+          link.href = url;
+          link.appendChild(brandLogo);
+        {% endif %}
+
+        {% if clickable or blockable %}
+            var overlay = document.getElementById('block-snippet-overlay-{{ snippet_id }}');
+        {% endif %}
+
+        {% if clickable %}
+          snippet.addEventListener('mouseover', function(e) {
+            overlay.classList.add('block-snippet-overlay-glow-{{ snippet_id }}');
+          });
+          snippet.addEventListener('mouseout', function(e) {
+            overlay.classList.remove('block-snippet-overlay-glow-{{ snippet_id }}');
+          });
+          snippet.addEventListener('click', function () {
+              var links = snippet.getElementsByTagName('a');
+              if (links.length) {
+                  links[0].click();
+              }
+          });
+        {% endif %}
+
+        {% if blockable %}
+          brandLogo.insertAdjacentHTML(
+            'afterbegin', '<div class="block-snippet-button block-snippet-button-{{ snippet_id }}"></div>');
+
+          var xbuttons = document.getElementsByClassName('block-snippet-button-{{ snippet_id }}');
+          var mouseover = function(event) {
+              overlay.classList.add('block-snippet-overlay-close-{{ snippet_id }}');
+              brandLogo.style.opacity = 0.6;
+          };
+          var mouseout = function(event) {
+              overlay.classList.remove('block-snippet-overlay-close-{{ snippet_id }}');
+              brandLogo.style.opacity = 1;
+          };
+          for (var k=0; k < xbuttons.length; k++) {
```

Comment:
Bit of an optimization is to cache the array length even though it is really small in this case.

```
for (var k = 0,l = xbuttons.length; k < l; k++) {
```